### PR TITLE
Build Time Speed-Ups

### PR DESCRIPTION
**Pull request by stephencelis**
I'll be tacking on some more stuff on this branch, but curious about CI speedups along the way.

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -20,22 +20,6 @@ let package = Package(
   products: [
     .executable(name: "Runner", targets: ["Runner"]),
     .executable(name: "Server", targets: ["Server"]),
-    .library(name: "Database", targets: ["Database"]),
```
Comment (stephencelis) [inline Package.swift]: This is only if other libraries wanna pull PointFree in and import frameworks...for now simpler to not maintain this list.

File: Sources/PointFree/About.swift [hunk 2]
```diff
@@ -10,7 +10,7 @@ import Views
 let aboutResponse: Middleware<StatusLineOpen, ResponseEnded, Tuple3<User?, SubscriberState, Route?>, Data> =
   writeStatus(.ok)
     >=> map(lower)
-    >>> _respond(
+    >>> respond(
```
Comment (stephencelis) [inline Sources/PointFree/About.swift]: You mentioned the underscored version can go away...

File: Sources/PointFree/Account/Change.swift [hunk 3]
```diff
@@ -31,10 +22,22 @@ let subscriptionChangeMiddleware: Middleware<
       )
 )
 
+private let requireUserAndPricingAndSeats
+  : MT<Tuple2<User?, Pricing?>, Tuple3<User, Int, Pricing>>
```
Comment (stephencelis) [inline Sources/PointFree/Account/Change.swift]: Using an `MT` type as a shorthand for describing middleware transformers...makes it slightly easier to break this stuff up.

File: Sources/PointFree/Account/Account.swift [hunk 4]
```diff
@@ -9,11 +9,11 @@ import Stripe
 import Tuple
 import Views
 
-let accountResponse: Middleware<StatusLineOpen, ResponseEnded, Tuple2<User?, SubscriberState>, Data> =
-  filterMap(require1 >>> pure, or: loginAndRedirect)
+let accountResponse//: M<Tuple2<User?, SubscriberState>>
```
Comment (mbrandonw) [inline Sources/PointFree/Account/Account.swift]: did you mean to drop the type and have this comment here?

File: Sources/PointFree/Account/Change.swift (hunk 3 above)
Comment (mbrandonw) [reply Sources/PointFree/Account/Change.swift]: I've actually got typealiases for this I've been using in other parts:

https://github.com/pointfreeco/pointfreeco/blob/master/Sources/PointFree/TODO.swift#L20-L21

File: Sources/PointFree/Admin/EpisodeCredit.swift [hunk 5]
```diff
@@ -21,15 +21,10 @@ let showEpisodeCreditsMiddleware: Middleware<
...
-let redeemEpisodeCreditMiddleware: Middleware<
-  StatusLineOpen,
-  ResponseEnded,
-  Tuple3<User, User.Id?, Int?>,
-  Data
-  > =
-  filterMap(
-      over2(fetchUser(id:)) >>> sequence2 >>> map(require2),
-      or: redirect(to: .admin(.episodeCredits(.show)), headersMiddleware: flash(.error, "Could not find that user."))
+let redeemEpisodeCreditMiddleware
```
Comment (mbrandonw) [inline Sources/PointFree/Admin/EpisodeCredit.swift]: is it not worth keeping the type here? does it not help at all?

File: Sources/PointFree/Admin/NewEpisodes.swift [hunk 6]
```diff
@@ -13,7 +13,7 @@ import Prelude
 import Styleguide
 import Tuple
 
-let showNewEpisodeEmailMiddleware: AppMiddleware<Prelude.Unit> =
+let showNewEpisodeEmailMiddleware: M<Prelude.Unit> =
```
Comment (mbrandonw) [inline Sources/PointFree/Admin/NewEpisodes.swift]: do we want to use `M` instead of `AppMiddleware`?

File: Package.swift (hunk 1 above)
Comment (mbrandonw) [reply Package.swift]: actually is this necessary to get targets in Xcode too? when I dont have these decelerations im not able to build these targets in isolation anymore

Comment (stephencelis) [reply Package.swift]: They all show up to me like this:

image

I'll address comments/merge if nothing's blocking, but we can revert this part if it's behaving weirdly.

File: Sources/PointFree/Account/Account.swift (hunk 4 above)
Comment (stephencelis) [reply Sources/PointFree/Account/Account.swift]: Most of the time, typing these middleware led to _slower_ builds (by 2x). I user comments to extract/simplify some of these. I'll delete this comment...

File: Sources/PointFree/Admin/EpisodeCredit.swift (hunk 5 above)
Comment (stephencelis) [reply Sources/PointFree/Admin/EpisodeCredit.swift]: Makes it worse somehow :/

File: Sources/PointFree/Admin/NewEpisodes.swift (hunk 6 above)
Comment (stephencelis) [reply Sources/PointFree/Admin/NewEpisodes.swift]: I didn't notice the overlap till I moved `M` and `MT` into TODO.swift. I figure this is all a temp situation till we rethink HttpPipeline, so cool with any names. Preference before merging?

Comment (mbrandonw) [reply Sources/PointFree/Admin/NewEpisodes.swift]: cool, let's keep as is then!

**Pull request merged**